pkg/codegen/internal/test/testdata/simple-enum-schema/schema.json

```diff
@@ -197,13 +197,6 @@
         }
       ]
     },
-    "plant:tree/v1:VarietiesSupported": {
-      "type": "array",
-      "items": {
-        "$ref": "#/types/plant:tree/v1:RubberTreeVariety"
-      },
-      "description": "The rubber tree varieties supported"
-    },
```

Comment:
Why did this get removed 🤔?

Reply:
Array type definitions aren't used by anything: only object and enum type definitions are processed by the schema binder. As such, the metaschema only allows object and enum type definitions in the `types` map. This schema was failing validation because of this array type definition, so I removed it.

Reply:
I was wondering about this the other day. CloudFormation schema does have explicit type definitions for array types. I modeled them as inline types for now. Alternatively, we could extend Pulumi schema to support those.

Reply:
Yeah, I've wondered about whether or not we need this. How is it used in CloudFormation?

@komalali your question reminded me that technically this is a breaking change for schema authors, as their previously compliant schemas will now be rejected by the binder. Do you think that we should allow these types, and maybe warn on them instead of just ignoring them?

Reply:
Example:

```json
    "CertificatePolicyList" : {
      "type" : "array",
      "items" : {
        "$ref" : "#/definitions/PolicyInformation"
      }
    },
    "PolicyInformation" : {
      "description" : "Structure that contains X.509 Policy information.",
      "type" : "object",
...
```